src/viam/sdk/rpc/dial.hpp

```diff
@@ -16,9 +16,16 @@ class ViamChannel {
    public:
     void close();
     ViamChannel(std::shared_ptr<GrpcChannel> channel, const char* path, void* runtime);
+
     static std::shared_ptr<ViamChannel> dial(const char* uri,
                                              const boost::optional<DialOptions>& options);
 
+    // allows for specifying that this dial is an initial connection attempt, indicating that
+    // initial connection options should be used.
+    static std::shared_ptr<ViamChannel> dial(const char* uri,
```

Comment:
I have something of a personal dislike for `bool` parameters like this, and it makes me wonder why this `initial` aspect isn't either just a setting on the `DialOptions`, or a differently named method.

My distaste arrises because at the call site, you end up like this:

```
ViamChannel::dial(my_uri, my_options, false);
```

What does `false` mean?

Whereas either of these makes the intention clear at the call site without needing to chase down the declaration / documentation for `dial`:

```
ViamChannel::dial(my_uri, my_options.as_initial());
```

or

```
ViamChannel::dial_initial(my_uri, my_options)
```